src/makie.jl

```diff
@@ -209,18 +225,293 @@ function new_cycle_theme()
+    ga = fig[1, 1] = GridLayout()
+    gb = fig[1, 2] = GridLayout()
+    gc = fig[1, 3] = GridLayout()
+    gd = fig[2, 1:3] = GridLayout()
+    gA = Axis(ga[1,1])
+    nested_sub_plot!(gb)
+    axsc = nested_sub_plot!(gc)
+    nested_sub_plot!(gd)
+
+    [hidedecorations!(axsc[i],grid=false,ticks=false) for i in 1:length(axsc)]
+
+    colgap!(gc, 5)
+    rowgap!(gc, 5)
+    rowsize!(fig.layout, 2, Auto(0.5))
+    colsize!(fig.layout, 1, Auto(0.5))
+    fig
+end
+
+function add_box_inset(fig;left=100,right=250,bottom = 200,top= 300, 
+        bgcolor=:grey90)
+    # https://discourse.julialang.org/t/makie-inset-axes-and-their-drawing-order/60987 # hide
+    inset_box = Axis(fig, bbox = BBox(left, right, bottom, top),
+        xticklabelsize = 12, yticklabelsize=12, backgroundcolor = bgcolor)
+    
+    # bring content upfront
+    translate!(inset_box.scene, 0, 0, 10)
+    elements = keys(inset_box.elements)
+    for ele in elements
+        if ele == :xaxis || ele == :yaxis
+            nothing
+        else 
+            translate!(inset_box.elements[ele], 0, 0, 9)
+        end
+    end
+    return inset_box
+end
+
+function add_axis_inset(; pos = fig[1,1], halign=0.1, valign=0.5, 
+        width = Relative(0.5), height= Relative(0.35), bgcolor=:lightgray)
+    
+    inset_box = Axis(pos, width = width, height = height, 
+        halign=halign, valign=valign, xticklabelsize = 12,yticklabelsize=12,
+        backgroundcolor = bgcolor)
+
+    # bring content upfront
+    translate!(inset_box.scene, 0, 0, 10)
+    elements = keys(inset_box.elements)
+    for ele in elements
+        if ele == :xaxis || ele == :yaxis
+            nothing
+        else 
+            translate!(inset_box.elements[ele], 0, 0, 9)
+        end
+    end
+    return inset_box
+end
+
+function figure_axis_inset()
+    CairoMakie.activate!() # hide
+    fig = Figure(resolution=(600,400))
+    ax = Axis(fig[1,1], backgroundcolor = :white)
+    inset_ax1 = add_axis_inset(; pos = fig[1,1], halign=0.1, valign=0.65,
+        width = Relative(0.3), height= Relative(0.3), bgcolor= :grey90)
+    inset_ax2 = add_axis_inset(; pos = fig[1,1], halign=1, valign=0.25,
+        width = Relative(0.25), height= Relative(0.3), bgcolor= (:white,0.65))
+
+    lines!(ax, 1:10)
+    lines!(inset_ax1, 1:10)
+    scatter!(inset_ax2, 1:10, color = :black)
+    fig
+end
+
+function figure_box_inset()
+    CairoMakie.activate!() # hide
+    fig = Figure(resolution=(600,400))
+    ax = Axis(fig[1,1], backgroundcolor = :white)
+    inset_ax1 = add_box_inset(fig; left=100,right=250,bottom = 200,top= 300, 
+        bgcolor = :grey90)
+    inset_ax2 = add_box_inset(fig; left=500,right=600,bottom = 100,top= 200, 
+        bgcolor = (:white,.65))
+
+    lines!(ax, 1:10)
+    lines!(inset_ax1, 1:10)
+    scatter!(inset_ax2, 1:10, color = :black)
+    fig
+end   
+
+# written by Josef Heinen from GR.jl
+"""
+    peaks([n=49])
+    
+    Returns a nonlinear function on a grid.  Useful for test cases.
```

Comment:
```suggestion
Return a nonlinear function on a grid.  Useful for test cases.
```
Julia convention is to use `return` instead of `returns` (this is somewhere in the Julia docs). Also removes the leading spaces. Leading spaces should only be at the function name (first line of the docstring)